Logger/LogFactory.cs

```diff
@@ -1,10 +1,34 @@
-﻿namespace Logger;
-
-public class LogFactory
-{
-    public BaseLogger CreateLogger(string className)
-    {
-        
-        return null;
-    }
-}
+﻿using System;
+using System.Diagnostics;
+
+namespace Logger;
+
+public class LogFactory
+{
+    public string? LogFilePath { get; set; }
+    public void ConfigureLogger(string filePath)
+    {
+        LogFilePath = filePath;
+    }
+    public BaseLogger? CreateLogger(string className, string callingClass)
+    {
+        BaseLogger? log;
+        switch (className)
+        {
+            case "ConsoleLogger":
+                log = new ConsoleLogger();
+                log.ClassName = className;
+                break;
+            case "FileLogger":
+                if (LogFilePath is null)
+                    return null;
+                log = new FileLogger(LogFilePath);
+                log.ClassName = className;

```

Comment:
```suggestion
                log = new FileLogger(LogFilePath)
                {
                    ClassName = className;
                }
```
This is how you use an object initializer